Zend/zend_vm_def.h

```diff
@@ -745,13 +748,35 @@ ZEND_VM_HELPER(zend_binary_assign_op_obj_helper, VAR|UNUSED|CV, CONST|TMPVAR|CV,
 				ZVAL_DEREF(zptr);
 				SEPARATE_ZVAL_NOREF(zptr);
 
-				binary_op(zptr, zptr, value);
-				if (UNEXPECTED(RETURN_VALUE_USED(opline))) {
+				if (OP2_TYPE == IS_CONST) {
+					prop_info = (zend_property_info*)CACHED_PTR_EX(cache_slot + 2);
+				} else {
+					prop_info = zend_object_fetch_property_type_info(Z_OBJCE_P(object), Z_STR_P(property), NULL);
```

Comment:
The only circumstance where a cache slot gets filled is when OP2_TYPE == IS_CONST and OP2 is a string.